eth2/types/src/slot_epoch_root.rs

```diff
@@ -51,11 +53,53 @@ impl Slot {
 }
 
 impl Epoch {
+    pub fn new(epoch: u64) -> Self {
+        Self(epoch)
+    }
+}
+
+impl EEIndex {
+    pub fn new(ee_index: u64) -> Self {
+        Self(ee_index)
+    }
+}
+
+impl Shard {
+    pub fn new(shard: u64) -> Self {
+        Self(shard)
+    }
+}
+
+impl ShardSlot {
     pub fn new(slot: u64) -> Self {
         Self(slot)
     }
 }
 
+/// Convert from a hex string to a Root
+impl TryFrom<&str> for Root {
```

Comment:
`FromStr` might be a more idiomatic approach, but I don't think this is incorrect.